doc/changelog/09-cli-tools/19370-compat-warning.rst

```diff
@@ -0,0 +1,6 @@
+- **Changed:**
+  The ``-compat`` :ref:`command line option <command-line-options>` is
+  now a warning rather than an error when the compatibility file
+  doesn't exist
```

Comment:
Maybe describe the reason for this change in case the users do not think of it by themselves.

BTW, what about a third changelog entry to note (or noting here) that compat files now silence deprecation warnings that were introduced since then?